client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpPipelineBuilder.java

```diff
@@ -128,12 +140,110 @@ public HttpPipelineBuilder withRequestPolicies(RequestPolicyFactory... requestPo
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withDecodingPolicy() {
+        return withRequestPolicy(new DecodingPolicyFactory());
+    }
+
+    /**
+     * Adds a RequestPolicy which sets the host on all outgoing requests.
+     * @param host The hostname to use in all outgoing requests.
+     * @return This HttpPipelineBuilder.
+     */
+    public HttpPipelineBuilder withHostPolicy(String host) {
+        return withRequestPolicy(new HostPolicyFactory(host));
+    }
+
+    /**
+     * Adds a RequestPolicy which logs all HTTP traffic using SLF4J.
+     * @param level The HTTP logging detail level.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withHttpLoggingPolicy(HttpLogDetailLevel level) {
+        return withHttpLoggingPolicy(level, false);
+    }
+
+    /**
+     * Adds a RequestPolicy which logs all HTTP traffic using SLF4J.
+     * @param level The HTTP logging detail level.
+     * @param prettyPrintJSON Whether or not to pretty print JSON message bodies.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withHttpLoggingPolicy(HttpLogDetailLevel level, boolean prettyPrintJSON) {
+        return withRequestPolicy(new HttpLoggingPolicyFactory(level, prettyPrintJSON));
+    }
+
+    /**
+     * Adds a RequestPolicy which adds proxy authentication headers to
+     * outgoing requests.
+     * @param username The username for authentication.
+     * @param password The password for authentication.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withProxyAuthenticationPolicy(String username, String password) {
+        return withRequestPolicy(new ProxyAuthenticationPolicyFactory(username, password));
+    }
+
+    /**
+     * Adds a RequestPolicy which adds a per-request ID to the
+     * "x-ms-client-request-id" header to outgoing requests.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withRequestIdPolicy() {
+        return withRequestPolicy(new RequestIdPolicyFactory());
+    }
+
+    /**
+     * Adds a RequestPolicy which retries a failed request up to the given
+     * number of times.
+     * @param maxRetries The maximum number of times to retry failed requests.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withRetryPolicy(int maxRetries) {
+        return withRequestPolicy(new RetryPolicyFactory(maxRetries));
+    }
+
+    /**
+     * Adds a RequestPolicy which fails a request if it does not complete by
+     * the time the given interval elapses.
+     * @param timeout The amount of time to wait before timing out a request.
+     * @param unit The unit of time associated with the timeout parameter.
+     * @return This HttpPipeline builder.
+     */
+    public HttpPipelineBuilder withTimeoutPolicy(long timeout, TimeUnit unit) {
+        return withRequestPolicy(new TimeoutPolicyFactory(timeout, unit));
```

Comment:
Now that we're using Java 8, should this be changed to use Duration?

Reply:
I guess the main reason for keeping it this way is alignment with the RxJava `.timeout()` API. If it were easy to translate the `Duration` to `(long, TimeUnit)` I'd be open to replacing or adding an overload.

Reply:
It would probably be easy enough to convert the Duration to milliseconds, but let's leave it for now.